### ISSUE TITLE
sdslideshow.py could skip progressive jpegs instead of throwing an exception

### ISSUE THREAD

**Issue by jose1711** [kevinjwalters/micropython-examples#1]
At this point whenever `jpeg.decode()` encounters a JPEG file with progressive encoding it throws an exception (and hence the whole slideshow stops). Ideally this is wrapped into `try.. except` block together with autoincrementing the cursor.

**Comment by kevinjwalters** [comment]
Yes, that's a good point, I imagine it's easy for a progressive one to sneak in and spoil the sequence. I should have some time this weekend to have a look.

**Comment by jose1711** [comment]
> Yes, that's a good point, I imagine it's easy for a progressive one to sneak in and spoil the sequence. I should have some time this weekend to have a look.

Thank you. That is true - speaking from a personal experience. And if you're running from AA batteries then being in an awake state is not the best idea either.

**Comment by kevinjwalters** [comment]
@jose1711  I've changed the code to skip past JPEGs that don't decode properly and to go into deep sleep if another exception happens to be thrown. Give it a go and let me know how it goes.

**Comment by jose1711** [comment]
Thank you, the code looks ok. I will test in the real life and report back in a few days. I am also thinking.. maybe an error condition could be indicated by an empty directory created on the card with the name of the problematic jpeg image in it? That way a user could work out candidates for fixing easily.

**Comment by kevinjwalters** [comment]
It prints to serial console now but that's only useful if it's attached to a computer. A single error log file or an error file per image as you suggest could be more useful, latter could be less write intensive. I'm also pondering a status screen on button D.
